packages/imjoy-api/src/viewer.ts

```diff
@@ -0,0 +1,22 @@
+import { Viewer } from '@itk-viewer/viewer/viewer.js';
+import { Viewport } from '@itk-viewer/viewer/viewport.js';
+import MultiscaleSpatialImage from '@itk-viewer/io/MultiscaleSpatialImage.js';
+
+type HyphaServer = {
+  registerService: (api: object) => any,
+  getService: (name: string) => any,
+  id: string,
+};
+
+export async function setup(server: HyphaServer, viewer: Viewer) {
+  const service = await server.registerService({
+    name: "itk_viewer",
+    id: "itk-viewer",
+    description: "Retrieve the viewer API",
+    viewer: () => viewer,
+    addViewport: (viewport: Viewport, name: string) => viewer.send({ type: 'addViewport', viewport: viewport, name: name }),
+    setImage: (image: MultiscaleSpatialImage, name?: string) => viewer.send({ type: 'setImage', image, name: name ?? 'image' }),
```

Comment:
Right now, perhaps all we need as an RPC API for Viewer actor is:
```
create: () => {
  viewerActor = createViewer()
 // returns nothing  (maybe an ID in future to distriquish bettween mulipble client side viewers?)
}
setImage: (zarrStore: ZarrStore, name: string)  => viewer.send({ type: 'setImage', image, name: name ?? 'image' }),
}
```

Reply:
Passing in an existing `Viewer` to setup does work if we want to continue to support that as well? Adding `create` is certainly an option as well though! I think I was just assuming that the `Viewer` may already exist on setup but that may not be the case.

Reply:
Good point.  Could drop `create` and assume each imjoy service has one to one mapping with a Viewer.`